graders/c/requirements.txt

```diff
@@ -1,3 +1,4 @@
+clang==18.1.8
```

Comment:
[PyPI has this package at 20.1.0](https://pypi.org/project/clang/), any reason why you're using an older version? Assuming this has to do with the clang version from apt (18.1.3), would it make more sense to also install the `python3-clang` apt package, since it would install the same Python dependency as this line but also match it to the `clang` package version itself (thus be less error-prone in regards to future development)?

Reply:
> would it make more sense to also install the `python3-clang` apt package

To clarify, I mean install it from apt and remove it from requirements.txt. The "also" here refers to "also from apt, like clang", not "also there in addition to here".

Reply:
Changed in 399333dd1f9a37df7f0a5a8e962d90b3fb7fb5ce.

Reply:
For completeness I tested mismatched versions of clang and the Python binding and they don't work, with an explicit check for version compatibility on start. Also, as discussed in another thread, we'll keep the pip version to keep the image at a reasonable size.

Reply:
Thanks for testing that. Do you think it's worth adding a comment here and/or in the Dockerfile to emphasize that the versions have to be kept in sync? Is there anything like a `clang-18` that we can install that's pinned to a specific version?

Reply:
Added a comment in 757b65da027cc98bd9c2d6ae0450e2dfcafc5b35.